packages/react/menu/src/Menu.tsx

```diff
@@ -450,24 +454,35 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
   // the last element in the DOM (beacuse of the `Portal`)
   useFocusGuards();
 
+  const isPointerMovingToSubmenu = React.useCallback((event: React.MouseEvent) => {
+    const isMovingTowards = pointerDirRef.current === pointerGraceIntentRef.current?.side;
+    return isMovingTowards && isPointerInGraceArea(event, pointerGraceIntentRef.current?.area);
+  }, []);
+
   return (
     <PortalWrapper>
       <ScrollLockWrapper>
         <MenuContentProvider
           searchRef={searchRef}
           onItemEnter={(event) => {
```

Comment:
Not sure how we missed this before, but this one needs `useCallback` now.

Reply:
Was wondering on this too, thought it might be a for a reason I was unaware of. Thanks for the reminder